src/inet/UDPEndPoint.cpp

```diff
@@ -434,10 +434,41 @@ void UDPEndPoint::Free()
 #if CHIP_SYSTEM_CONFIG_USE_LWIP
     DeferredFree(kReleaseDeferralErrorTactic_Die);
 #else  // !CHIP_SYSTEM_CONFIG_USE_LWIP
-    Release();
+    UDPEndPoint::sPool.ReleaseObject(this);
 #endif // !CHIP_SYSTEM_CONFIG_USE_LWIP
 }
 
+#if CHIP_SYSTEM_CONFIG_USE_LWIP
+void UDPEndPoint::DeferredFree(ReleaseDeferralErrorTactic aTactic)
```

Comment:
DeferredRelease needs to be used for all Lwip implementations, we should move this back to SystemPool.